cert/spec.go

```diff
@@ -483,16 +475,56 @@ func (spec *Spec) TakeAction(changeType string) error {
 // The maximum number of attempts before giving up.
 const maxAttempts = 5
 
-// renewPKI Try to update the on disk PKI content with a fresh CA/cert as needed
-func (spec *Spec) renewPKI(ca *x509.Certificate) error {
-	metrics.SpecRefreshCount.WithLabelValues(spec.Path).Inc()
-	failed := true
+// writePKIToDisk writes the given PKI materials to disk, returning errors if anything occurs.
+// CA must always be passed, but is optionally written if the spec defines an on disk CA.
+// keypair can be nil if the spec does not write a cert/key- if it's just used for tracking the CA.
+func (spec *Spec) writePKIToDisk(ca *x509.Certificate, keyPair *tls.Certificate) (err error) {
```

Comment:
I strongly prefer specifying only a type to return, rather than a named return variable. This is subjective, of course, so feel free to disregard, but I feel that it makes the control flow a lot more readable to just `return err` rather than a naked return. It especially improves readability for engineers less familiar with Go, but even I have to give something like this a second look and retain an extra thing in my memory to parse this.